packages/@tailwindcss-upgrade/src/utils/dimension.ts

```diff
@@ -0,0 +1,18 @@
+import { DefaultMap } from '../../../tailwindcss/src/utils/default-map'
+
+// Parse a dimension such as `64rem` into `[64, 'rem']`.
+export const dimensions = new DefaultMap((input) => {
+  let match = /^(?<value>-?(\d*)?\.?\d+)(?<unit>.*)$/.exec(input)
```

Comment:
I think something like this might be a bit better

```suggestion
  let match = /^(?<value>-?(?:\d*\.)?\d+)(?<unit>[a-z]+|%)$/i.exec(input)
```